session_state.go

```diff
@@ -157,11 +157,11 @@ func (sm *stateMachine) CheckSessionTime(session *session, now time.Time) {
 
 func (sm *stateMachine) CheckResetTime(session *session, now time.Time) {
 	if session.EnableResetSeqTime {
-		if session.ResetSeqTime.Hour() == now.Hour() &&
-			session.ResetSeqTime.Minute() == now.Minute() &&
-			session.ResetSeqTime.Second() == now.Second() {
+		// If we have crossed the reset time boundary in between checks, we send the reset
+		if session.LastCheckedResetSeqTime.Before(session.ResetSeqTime) && now.After(session.ResetSeqTime) && session.stateMachine.State.IsConnected() {
```

Comment:
This will not work. The session.ResetSeqTime, when parsed with the short form string in the session factory, will correspond to the time of the parse only. It will not reflect "today's" reset seq time.

In addition, the last checked reset seq time will be initialized to zero, so this method needs to ensure that the last checked reset seq time must have been set previously before passing this condition